entity-service/src/test/kotlin/com/egm/stellio/entity/web/EntityOperationHandlerTests.kt

```diff
@@ -178,12 +178,116 @@ class EntityOperationHandlerTests {
             )
     }
 
+    @Test
+    fun `upsert batch entity should return a 200 if JSON-LD payload is correct`() {
+        val jsonLdFile = ClassPathResource("/ngsild/hcmr/HCMR_test_file.json")
+        val entitiesIds = arrayListOf(
+            "urn:ngsi-ld:Sensor:HCMR-AQUABOX1temperature",
+            "urn:ngsi-ld:Sensor:HCMR-AQUABOX1dissolvedOxygen",
+            "urn:ngsi-ld:Device:HCMR-AQUABOX1"
+        )
+
+        val existingEntities = mockk<List<ExpandedEntity>>()
+        val nonExistingEntities = mockk<List<ExpandedEntity>>()
+
+        every { entityOperationService.splitEntitiesByExistence(any()) } returns Pair(
+            existingEntities,
+            nonExistingEntities
+        )
+        every { entityOperationService.create(nonExistingEntities) } returns BatchOperationResult(
+            arrayListOf(),
+            arrayListOf()
+        )
+        every { entityOperationService.update(existingEntities) } returns BatchOperationResult(
+            entitiesIds,
+            arrayListOf()
+        )
+        webClient.post()
+            .uri("/ngsi-ld/v1/entityOperations/upsert")
+            .header("Link", "<$aquacContext>; rel=http://www.w3.org/ns/json-ld#context; type=application/ld+json")
+            .bodyValue(jsonLdFile)
+            .exchange()
+            .expectStatus().isOk
+            .expectBody().json(
+                "{\n" +
```

Comment:
Use multi-line string so that we don't propagate more of these hard to read string concatenations :)